doc/src/contracts/multisig_prover.md

```diff
@@ -62,59 +112,113 @@ Prover-->>-Relayer: returns GetProofResponse
+7. Multisig emits event `SigningStarted` indicating a new multisig session has started
+8. Multisig triggers a reply in Prover returning the newly created session ID which is then stored with the batch for reference
+9. Prover contract emits event `ProofUnderConstruction` which includes the ID of the proof being constructed.
+10. Signers submit their signatures until threshold is reached
+11. Multisig emits event indicating the multisig session has been completed
+12. Relayer queries Prover for the proof, using the proof ID
+13. Prover queries Multisig for the multisig session, using the session ID
+14. Multisig replies with the multisig state, the list of collected signatures so far and the snapshot of participants.
+15. If the Multisig state is `Completed`, the Prover finalizes constructing the proof and returns the `GetProofResponse` struct which includes the proof itself and the data to be sent to the destination gateway. If the state is not completed, the Prover returns the `GetProofResponse` struct with the `status` field set to `Pending`.
+
+
+## Update and confirm WorkerSet graph
 
-#[derive(QueryResponses)]
-pub enum QueryMsg {
-    #[returns(GetProofResponse)]
-    GetProof { multisig_session_id: Uint64 },
-}
+```mermaid
+graph TD
 
-pub enum ProofStatus {
-    Pending,
-    Completed { execute_data: HexBinary }, // encoded data and proof sent to destination gateway
-}
+r[Relayer]
+subgraph Axelar
+b[Prover]
+v[Voting Verifier]
+m[Multisig]
+s[Service Registry]
+end
 
-pub struct GetProofResponse {
-    pub multisig_session_id: Uint64,
-    pub message_ids: Vec<String>,
-    pub data: Data,
-    pub status: ProofStatus,
-}
+r--UpdateWorkerSet-->b
+b--GetActiveWorkers-->s
+b--RegisterWorkerSet-->m
+r--ConfirmWorkerSet-->b
+b--IsWorkerSetConfirmed-->v
 ```
 
-## Events
 
-```Rust
-pub enum Event {
-    ProofUnderConstruction {
-        multisig_session_id: Uint64,
-    },
-    SnapshotRotated {
-        key_id: String,
-        snapshot: Snapshot,
-        pub_keys: HashMap<String, HexBinary>,
-    },
-}
+## Update and confirm WorkerSet sequence diagram
+
+```mermaid
+sequenceDiagram
+autonumber
+participant External Gateway
+participant Relayer
+box LightYellow Axelar
+participant Prover
+participant Service Registry
+participant Voting Verifier
+participant Multisig
+end
+actor Signers
+Relayer->>+Prover: ExecuteMsg::UpdateWorkerSet
+alt no WorkerSet stored
+  Prover->>+Service Registry: QueryMsg::GetActiveWorkers
+  Service Registry-->>-Prover: save new WorkerSet as current WorkerSet
+  Prover->>+Multisig: ExecuteMsg::RegisterWorkerSet
+else existing WorkerSet stored
+  Prover->>+Service Registry: QueryMsg::GetActiveWorkers
+  Service Registry-->>-Prover: save new WorkerSet as next WorkerSet
+  Prover->>+Multisig: ExecuteMsg::StartSigningSession (for operatorship transferred message)
+  Multisig-->>Signers: emit SigningStarted event
+  Multisig->>-Prover: reply with session ID
+  Prover-->>Relayer: emit ProofUnderConstruction event
```

Comment:
You can omit these 3 lines, they are described in the previous diagram. This flow is more about the whole worker set update process. The nuances of signing proofs is best left to the previous diagram, to keep this one readable.